### PR TITLE
New menu structure 2: Design pages

### PR DESCRIPTION
**Pull request by PatrykSzwer**
For #283 

Note 1: First I added typography page but then in latest commit I removed its files, because TBH I didn't know what descriptions/examples put there. If you disagree and have some ideas what to put there, please share with me.

Note 2: In this PR I started some changes that I would like to make in entire solution in separated "Refactor kitchensink structure" PR. Those changes are e.g regions in Program.cs for better readability and folders structure that represent same structure as navigation menu.

### CODE REVIEW

File: src/KitchenSink/Program.cs [hunk 1]
```diff
@@ -240,7 +250,7 @@ static void Main()
         {
             var master = GetMasterPageFromSession();
 
-            if (master.CurrentPage != null && master.CurrentPage.GetType().Equals(typeof(T)))
+            if (master.CurrentPage != null && master.CurrentPage.GetType() == typeof(T))
```
Comment (alshakero) [inline src/KitchenSink/Program.cs]: Maybe its best to use `Object.ReferenceEquals`? see https://msdn.microsoft.com/en-us/library/system.object.gettype(v=vs.110).aspx

File: src/KitchenSink/wwwroot/KitchenSink/Design/Alerts.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
+﻿<link rel="preload" as="style" href="/sys/uniform.css/uniform.css">
+
+<template>
+    <dom-bind>
+        <template is="dom-bind">
+            <h1 slot="kitchensink/alerts-heading">Alerts</h1>
+            <p slot="kitchensink/alerts-description">Uniform provides couple of alerts classes to mark importance and character of specific messages.</p>
```
Comment (alshakero) [inline src/KitchenSink/wwwroot/KitchenSink/Design/Alerts.html]: ~~couple of alerts classes to mark~~ → a few alert classes to indicate

File: src/KitchenSink/wwwroot/KitchenSink/Design/Alerts.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            <p slot="kitchensink/alerts-warning-description"><code>.uni-alert-warning</code> can be used to indicate warning messages.</p>
+            <p slot="kitchensink/alerts-warning-alert">The cafeteria will be closed for maintanance today between 08:00 and 10:00 UTC.</p>
+            
+            <h2 slot="kitchensink/alerts-success-title">Success</h2>
+            <p slot="kitchensink/alerts-success-description"><code>.uni-alert-success</code> can be used to indicate success messages.</p>
+            <p slot="kitchensink/alerts-success-alert">You successfuly cooked a pizza.</p>
+            
+            <h2 slot="kitchensink/alerts-danger-title">Danger</h2>
+            <p slot="kitchensink/alerts-danger-description"><code>.uni-alert-danger</code> can be used to indicate danger messages.</p>
+            <p slot="kitchensink/alerts-danger-alert">The cafeteria run out off pizza!</p>
```
Comment (alshakero) [inline src/KitchenSink/wwwroot/KitchenSink/Design/Alerts.html]: ~~off~~ → of

File: src/KitchenSink/wwwroot/KitchenSink/Design/Alerts.html [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            <p slot="kitchensink/alerts-info-description"><code>.uni-alert-info</code> can be used to indicate information messages.</p>
+            <p slot="kitchensink/alerts-info-alert">The cafeteria serves pizza for lunch!</p>
+            
+            <h2 slot="kitchensink/alerts-warning-title">Warning</h2>
+            <p slot="kitchensink/alerts-warning-description"><code>.uni-alert-warning</code> can be used to indicate warning messages.</p>
+            <p slot="kitchensink/alerts-warning-alert">The cafeteria will be closed for maintanance today between 08:00 and 10:00 UTC.</p>
+            
+            <h2 slot="kitchensink/alerts-success-title">Success</h2>
+            <p slot="kitchensink/alerts-success-description"><code>.uni-alert-success</code> can be used to indicate success messages.</p>
+            <p slot="kitchensink/alerts-success-alert">You successfuly cooked a pizza.</p>
```
Comment (alshakero) [inline src/KitchenSink/wwwroot/KitchenSink/Design/Alerts.html]: cooked → baked 

You don't cook pizza 🤣

File: src/KitchenSink/wwwroot/KitchenSink/Design/Title.html [hunk 5]
```diff
@@ -0,0 +1,38 @@
+﻿<link rel="preload" as="style" href="/sys/uniform.css/uniform.css">
+
+<template>
+    <dom-bind>
+        <template is="dom-bind">
+            <h1 slot="kitchensink/title-heading">Title</h1>
+            <p slot="kitchensink/title-description">Use <code>.uni-title</code> class inside <code>.uni-section</code> or <code>.uni-card</code> to apply consistently formatted page or card title. Note that <code>.uni-title</code> overwrites the font style of the slotted element.</p>
+            <p slot="kitchensink/title-description"> We recommend to use semantic HTML elements for the title content, such as <code>h1</code> for the page title and <code>h2</code> for the card title.</p>
```
Comment (alshakero) [inline src/KitchenSink/wwwroot/KitchenSink/Design/Title.html]: to use → using

File: src/KitchenSink/Program.cs (hunk 1 above)
Comment (PatrykSzwer) [reply src/KitchenSink/Program.cs]: It's not better, in this case, It's the same.

**Comment by PatrykSzwer** [comment]
Thanks for review @alshakero!
I'll wait with merge until @tomalec says his opinion, especially about this typography page.

File: src/KitchenSink/Program.cs [hunk 6]
```diff
@@ -168,12 +188,7 @@ static void Main()
                 {
                     var master = session.Store[nameof(MasterPage)] as MasterPage;
 
-                    if (master == null)
-                    {
-                        return;
-                    }
-
-                    var page = master.CurrentPage as FileUploadPage;
+                    var page = master?.CurrentPage as FileUploadPage;
```
Comment (tomalec) [inline src/KitchenSink/Program.cs]: `?.` ❤️  I can't wait to have it in JS

File: src/KitchenSink/wwwroot/KitchenSink/Design/LeftNavLayout.html [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+                        <slot name="kitchensink/leftnavlayout-navigation"></slot>
+                    </nav>
+                    <div class="uni-section-primary">
+                        <slot name="kitchensink/leftnavlayout-navigation-content"></slot>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </template>
+</template>
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/Design/LeftNavLayout.html]: Having a separate page with this, I think we can remove redundant info in main page.

**Comment by tomalec** [comment]
Speaking of "typography", maybe we can have just a link "Native elements" to https://starcounter.github.io/underwear.css/ instead?

File: src/KitchenSink/Program.cs (hunk 6 above)
Comment (PatrykSzwer) [reply src/KitchenSink/Program.cs]: One of many miracles provided by all mighty C# 📿 🤴

File: src/KitchenSink/wwwroot/KitchenSink/Design/LeftNavLayout.html (hunk 7 above)
Comment (PatrykSzwer) [reply src/KitchenSink/wwwroot/KitchenSink/Design/LeftNavLayout.html]: Done. I also added short info about app-shell to leftNavLayout page.

File: src/KitchenSink/Program.cs (hunk 6 above)
Comment (tomalec) [reply src/KitchenSink/Program.cs]: https://github.com/tc39/proposal-optional-chaining

**Pull request merged**